node/client/src/lib.rs

```diff
@@ -349,15 +365,26 @@ impl Client {
             };
         }
 
-        self.pending_shard_blocks
-            .write()
-            .expect(POISONED_LOCK_ERR)
-            .insert(shard_block);
+        self.pending_shard_blocks.write().expect(POISONED_LOCK_ERR).insert(shard_block);
         self.pending_beacon_blocks.write().expect(POISONED_LOCK_ERR).insert(beacon_block);
-        let best_block_hash = self.beacon_chain.chain.best_hash();
+        let best_block_hash = self.beacon_client.chain.best_hash();
 
+        let new_best_block = self.beacon_client.chain.best_block();
+        self.try_apply_pending_blocks();
+
+        if new_best_block.block_hash() == best_block_hash {
```

Comment:
It seems that this equality always holds. Should `try_apply_pending_blocks` be called before getting `new_best_block`?

Reply:
Good catch. I was wondering why the tests are breaking..